### PR TITLE
[TACHYON-1156] Partition tachyon.thrift into per service files and add comments

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://tachyon.atlassian.net/browse/TACHYON-1156

This PR does a few things:
- separate tachyon.thrift into multiple files
- renames RPCs `renameFile` to `rename` and `deleteFile` to `remove` (unfortunately `delete` is a reserved keyword in Thrift)
- sorts methods in FileSystemServiceHandler alphabetically

### CODE REVIEW

File: bin/tachyon [hunk 1]
```diff
@@ -281,7 +281,9 @@ elif [[ ${COMMAND} == "copyDir" ]]; then
   exit 0
 elif [[ ${COMMAND} == "thriftGen" ]]; then
   rm -rf ${bin}/../common/src/main/java/tachyon/thrift
-  thrift --gen java -out ${bin}/../common/src/main/java/. ${bin}/../common/src/thrift/tachyon.thrift
+  for service in common exception block_master file_master lineage_master raw_table_master worker; do
```
Comment (calvinjia) [inline bin/tachyon]: Is there anyway to make this less hard coded?

File: common/src/thrift/file_master.thrift [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+   *
+   * NOTE: Unfortunately, the method cannot be called "delete" as that is a reserved Thrift keyword.
+   */
+  bool remove(1: i64 id, 2: bool recursive)
+    throws (1: exception.TachyonTException e)
+
+  /*
+   * Renames a file or a directory.
+   */
+  bool rename(1: i64 fileId, 2: string dstPath)
```
Comment (calvinjia) [inline common/src/thrift/file_master.thrift]: Should other methods also have their qualifier stripped (ie. `persistFile` -> `persist`)? Though this should be done in another PR. Will this work out with the multiplexed service handler? (we may have multiple `create`s)

**Comment by calvinjia** [comment]
@jsimsa Thanks for the patch, the thrift file structure looks much better now. I would prefer to not have any more renames in this PR.

LGTM

**Comment by yupeng9** [comment]
Thanks for the reorganizing the thrift files!

LGTM

File: bin/tachyon (hunk 1 above)
Comment (jsimsa) [inline bin/tachyon]: Done.

File: common/src/thrift/file_master.thrift (hunk 2 above)
Comment (jsimsa) [inline common/src/thrift/file_master.thrift]: I think that with the multiplexed service handler, method names will need to be unique. I think persistFile can stay as is, because the argument will always be a file (not a directory).

I would like to clean the inode vs. file vs. directory vs. folder naming throughout are code base, but I will first write a proposal for that to make sure everyone is on the same page.

**Comment by jsimsa** [comment]
Jenkins, test this please.

File: clients/unshaded/src/main/java/tachyon/client/FileSystemMasterClient.java [hunk 3]
```diff
@@ -272,21 +272,21 @@ public synchronized void completeFile(long fileId) throws IOException, TachyonEx
...
    */
-  public synchronized boolean deleteFile(long fileId, boolean recursive) throws IOException,
+  public synchronized boolean delete(long id, boolean recursive) throws IOException,
       TachyonException {
     int retry = 0;
     while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
       connect();
       try {
-        return mClient.deleteFile(fileId, recursive);
+        return mClient.remove(id, recursive);
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/FileSystemMasterClient.java]: why rename this from `deleteFile` to `remove` rather than `delete`?

Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/FileSystemMasterClient.java]: see the notes `NOTE: Unfortunately, the method cannot be called "delete" as that is a reserved Thrift keyword.`

Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/FileSystemMasterClient.java]: Because unfortunatley `delete` is a reserved keyword in Thrift.

**Comment by apc999** [comment]
+1 LGTM
